algorithms-miscellaneous-3/src/test/java/com/baeldung/algorithms/breadthfirstsearch/BreadthFirstSearchAlgorithmUnitTest.java

```diff
@@ -0,0 +1,85 @@
+package com.baeldung.algorithms.breadthfirstsearch;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class BreadthFirstSearchAlgorithmUnitTest {
+
+    private Tree<Integer> root;
+    private Tree<Integer> rootFirstChild;
+    private Tree<Integer> depthMostChild;
+    private Tree<Integer> rootSecondChild;
+
+    private Node<Integer> start;
+    private Node<Integer> firstNeighbour;
+    private Node<Integer> firstNeighbourNeighbour;
+    private Node<Integer> secondNeighbour;
+
+    @BeforeEach
+    void beforeEach() {
+        initTree();
+        initNode();
```

Comment:
Since we're only using one or the other in each test, let's just initialize the appropriate one when we need it instead of always initializing both.